src/apps/bridge/sensor_drivers/borealisSensor.cpp

```diff
@@ -287,6 +292,57 @@ BmErr BorealisSensor::encode_sample(void *data, uint32_t sample_index,
   return err;
 }
 
+/*!
+ @brief Log Hydrotwin Data To Spotter SD Card
+
+ @param data Data to log to SD card
+ @param data_len Length of data
+ @param type BorealisSubscription_t, must be HYDROTWIN_LDR or HYDROTWIN_HDR
+
+ @return BmOK on success, BMErr on failure
+ */
+BmErr BorealisSensor::hydrotwinSendSpotterLog(const uint8_t *data, uint16_t data_len,
+                                              BorealisSubscriptions_t type) {
+  BmErr err = BmEINVAL;
+  size_t encoded_data_len = 0;
+
+  mbedtls_base64_encode(NULL, 0, &encoded_data_len, (const unsigned char *)data, data_len);
+
+  // Base64 encoded data
+  if (encoded_data_len) {
+    uint8_t *encoded_data = static_cast<uint8_t *>(bm_malloc(encoded_data_len));
+    err = BmENOMEM;
+
+    if (encoded_data) {
+      mbedtls_base64_encode(encoded_data, encoded_data_len, &encoded_data_len,
+                            (const unsigned char *)data, data_len);
```

Comment:
It would be safer to have a separate output variable, rather than using encoded_data_len twice. This should be fine in most cases, but I wouldn't want hard-to-understand, under-the-hood, compiler optimization shenanigans to screw that up.